File: agent-workspace-sdk/tests/smoke.py
"""Smoke test — requires aw-api running on http://localhost:4000 without auth."""

import asyncio
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))

from agent_workspace import WorkspaceClient, TaskConflictError


async def main() -> None:
    client = WorkspaceClient(base_url="http://localhost:4000", agent_id="smoke-test-agent")

    print("1. Registering agent...")
    agent = await client.register(name="Smoke Test Agent", role="tester", capabilities=["smoke"])
    print(f"   OK: {agent.id}")

    print("2. check_in + enrichment...")
    async with client.session() as session:
        print(f"   session.id      = {session.id}")
        print(f"   pending_tasks   = {len(session.pending_tasks)}")
        print(f"   inbox           = {len(session.inbox)}")
        print(f"   pending_handoffs= {len(session.pending_handoffs)}")

        print("3. create_task...")
        task = await session.create_task(
            "Smoke task", "Created by smoke test", kind="custom:smoke", priority="low"
        )
        print(f"   task.id = {task.id}")

        print("4. claim_task...")
        claimed = await session.claim_task(str(task.id))
        print(f"   status = {claimed.status}")

        print("5. conflict detection (claim again)...")
        try:
            await session.claim_task(str(task.id))
            print("   FAIL: expected TaskConflictError")
        except TaskConflictError:
            print("   OK: TaskConflictError raised correctly")

        print("6. update_task_status → done...")
        updated = await session.update_task_status(str(task.id), "done")
        print(f"   status = {updated.status}")

        print("7. send_message...")
        await session.send_message(
            to_agent_id="smoke-test-agent",
            kind="status_update",
            payload={"msg": "smoke complete"},
        )
        print("   OK")

        print("8. list_inbox...")
        items = await session.list_inbox()
        print(f"   {len(items)} item(s)")

        print("9. workspace summary...")
        summary = await client.get_summary()
        print(f"   active_sessions={summary.active_sessions}")

    print("\n✅ All smoke tests passed")


asyncio.run(main())

File: agent-workspace-sdk-ts/tests/smoke.ts
// Smoke test — requires aw-api running on http://localhost:4000 without auth
import { WorkspaceClient, TaskConflictError } from "../src/index.js";

const client = new WorkspaceClient({
  baseUrl: "http://localhost:4000",
  agentId: "smoke-test-agent-ts",
});

console.log("1. Registering agent...");
const agent = await client.register({
  name: "Smoke Test Agent TS",
  role: "tester",
  capabilities: ["smoke"],
});
console.log(`   OK: ${agent.id}`);

console.log("2. checkIn + enrichment...");
await client.withSession(async (session) => {
  console.log(`   session.id       = ${session.id}`);
  console.log(`   pendingTasks     = ${session.pendingTasks.length}`);
  console.log(`   inbox            = ${session.inbox.length}`);
  console.log(`   pendingHandoffs  = ${session.pendingHandoffs.length}`);

  console.log("3. createTask...");
  const task = await session.createTask(
    "Smoke task TS",
    "Created by smoke test",
    "custom:smoke",
    "low",
  );
  console.log(`   task.id = ${task.id}`);

  console.log("4. claimTask...");
  const claimed = await session.claimTask(task.id!);
  console.log(`   status = ${claimed.status}`);

  console.log("5. conflict detection (claim again)...");
  try {
    await session.claimTask(task.id!);
    console.error("   FAIL: expected TaskConflictError");
    process.exit(1);
  } catch (err) {
    if (err instanceof TaskConflictError) {
      console.log("   OK: TaskConflictError raised correctly");
    } else {
      throw err;
    }
  }

  console.log("6. updateTaskStatus → done...");
  const updated = await session.updateTaskStatus(task.id!, "done");
  console.log(`   status = ${updated.status}`);

  console.log("7. sendMessage...");
  await session.sendMessage({
    toAgentId: "smoke-test-agent-ts",
    kind: "status_update",
    payload: { msg: "smoke complete" },
  });
  console.log("   OK");

  console.log("8. listInbox...");
  const items = await session.listInbox();
  console.log(`   ${items.length} item(s)`);

  console.log("9. workspace summary...");
  const summary = await client.getSummary();
  console.log(`   activeSessions=${summary.activeSessions}`);
});

console.log("\n✅ All TypeScript smoke tests passed");
